### ISSUE TITLE
errors on vagrant up

### ISSUE THREAD

**Issue by mateomtb** [Automattic/vip-go-quickstart#16]
```
mateo@machine:~/Repo/vip-go-qs (master)$ vagrant provision
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* The host path of the shared folder is missing: ./go-client-repo/languages
* The host path of the shared folder is missing: ./go-client-repo/vip-config
```

**Comment by philipjohn** [comment]
Hi Mateo,

Could you take me through all the steps that led to this error, please?

It looks like the two synced folders are missing, but I wouldn't expect it to cause as big a fail. I'm also aware your repo does have the `vip-config` folder.

Thanks

**Comment by mateomtb** [comment]
Ugh my fault forgot to add all the synch folders, was up way too late. Now just sorting out why it's not completing. Im clearing out all my old quickstart and vvv. I had them all shut down and hosts cleared but this vagrant up is not adding the host entries. The box fires up and i can ssh to it.

```
==> default: Running provisioner: puppet...
==> default: Running Puppet with default.pp...
==> default: stdin: is not a tty
==> default: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
==> default: Error: Cannot create a location reference without attaching to a virtual host at /tmp/vagrant-puppet/modules-04b25187997451ff8bd8d764087ef8fc/nginx/manifests/resource/location.pp:309 on node go-vip.local
==> default: Error: Cannot create a location reference without attaching to a virtual host at /tmp/vagrant-puppet/modules-04b25187997451ff8bd8d764087ef8fc/nginx/manifests/resource/location.pp:309 on node go-vip.local
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by mateomtb** [comment]
Just wanted to confirm a few others I work with have tried this and are having the same error

```
=> default: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
==> default: Error: Cannot create a location reference without attaching to a virtual host at /tmp/vagrant-puppet/modules-bd88de2fbac17777073fb9dfed3fa486/nginx/manifests/resource/location.pp:309 on node go-vip.local
==> default: Error: Cannot create a location reference without attaching to a virtual host at /tmp/vagrant-puppet/modules-bd88de2fbac17777073fb9dfed3fa486/nginx/manifests/resource/location.pp:309 on node go-vip.local
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by ethitter** [comment]
Hi @mateomtb! You've gotten caught up by issue #15, the outdated readme. After #13 was merged, an init script was introduced, `qs-init.sh`, that the Vagrantfile is dependent upon. Until I get the readme updated (soon, just after the `feature/content-and-confirmations` branch is merged), you can refer to #13 and its example of how to use the helper script.

**Comment by mateomtb** [comment]
thank you so much i'm getting closer, not sure what i'm doing wrong. I just give a slug and pass the url to my repo correct? I know the repo has a vip-config in it. Languages is not in there. There is no actual VM being created.

```
mateo@machine:~/Repo/vip-go-qs (master)$ ./qs-init.sh --client vip-go-test.local --git-repo git@github.com:dfmedia/wp-mason.git [--up]

Preparing to initialize the VIP Go Quickstart environment...

1) Parsing arguments...

CLIENT: vip-go-test.local
GIT REMOTE: git@github.com:dfmedia/wp-mason.git
VAGRANT UP REQUESTED: 0

2) Loading new client code...

Cloning into 'go-client-repo-new'...
remote: Counting objects: 6652, done.
remote: Compressing objects: 100% (62/62), done.
remote: Total 6652 (delta 38), reused 0 (delta 0), pack-reused 6590
Receiving objects: 100% (6652/6652), 2.59 MiB | 2.20 MiB/s, done.
Resolving deltas: 100% (4438/4438), done.
Checking connectivity... done.
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvX] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvX] source_file ... target_directory
mkdir: ./go-client-repo: No such file or directory

3) Starting to re-provision VM...

There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* The host path of the shared folder is missing: ./go-client-repo/languages
* The host path of the shared folder is missing: ./go-client-repo/vip-config



4) DONE!
Your VIP Go Quickstart is ready at http://go-vip.local/
Synchronized git checkout is at ./go-client-code/
mateo@machine:~/Repo/vip-go-qs (master)$
```

**Comment by ethitter** [comment]
Looks like two different problems, @mateomtb.

First, can you confirm that the `dfmedia/wp-mason` repo includes the `vip-config/vip-config.php` file/folder that's included in the [skeleton repo](https://github.com/Automattic/vip-skeleton)? One of the errors indicates that the file is missing, but it is a required part of the Go repo.

Second, can you switch to the `feature/content-and-confirmations` branch and try again, using the command below? The readme being worked on in #18 will clarify how the init script is used; in particular, the _client_ value isn't used as the domain and can't be in that format, and the square brackets indicate optional parameters but aren't included in the commands. In the interest of speed in re-initializing with a new git repo, the VM is always served at http://go-vip.local/.

`./qs-init.sh --client vip-go-test-local --git-repo git@github.com:dfmedia/wp-mason.git --up`

**Comment by mateomtb** [comment]
THANK YOU! Yes there has been a `vip-config/vip-config.php` in that repo all along. And good news that branch worked! I like how this works, very nice. I have other questions in regards to our build going to Go that uses 2 WP sites (and more in the future) that talk to each other via REST API. Trying to figure out how to do this in local dev without multisite, that's how we have been working up till now. I see how this is intended to switch bases but i need 2 instances running at once. Is it appropriate to address that here or should I start a new issue?

```
4) DONE!
Your VIP Go Quickstart is ready at http://go-vip.local/
Synchronized git checkout is at ./go-client-code/
mateo@machine:~/Repo/vip-go-qs (feature/content-and-confirmations)$
```

**Comment by ethitter** [comment]
@mateomtb, best to open a new issue. The initial implementation doesn't expressly support multisite, but subdirectory multisite should work with the necessary additions to `vip-config/vip-config.php` (these shouldn't be committed to the production repos, though). This is entirely untested at the moment, but given that we're supporting subdirectory multisite on Go, the Quickstart should match. Thanks for raising this.

Also, note that since the domain is hardcoded (for reasons discussed in #13) and Zeroconf is used, only one instance of the VM, and its one Go git repo, is supported at one time. Given your specific needs, which aren't likely to be unique, we'll need to revisit this decision. Mind opening an issue for that as well, describing your particular use case?

**Comment by ethitter** [comment]
@mateomtb, we've merged several updates to master today. Could you try again and let us know if the results are any better? Thanks!
